check_api/src/main/java/com/google/errorprone/SuppressionInfo.java

```diff
@@ -82,18 +164,29 @@ private static boolean isGenerated(Symbol sym) {
   public SuppressedState suppressedState(
       Suppressible suppressible, boolean suppressedInGeneratedCode, VisitorState state) {
     if (inGeneratedCode && suppressedInGeneratedCode) {
-      return SuppressedState.SUPPRESSED;
+      return new Suppressed(null);
     }
     if (suppressible.supportsSuppressWarnings()
         && (suppressWarningsStrings.contains("all")
             || !Collections.disjoint(suppressible.allNames(), suppressWarningsStrings))) {
-      return SuppressedState.SUPPRESSED;
+      String name;
+      if (suppressWarningsStrings.contains("all")) {
+        name = "all";
+      } else {
+        // find the first name that suppresses this check
+        name =
+            suppressible.allNames().stream()
+                .filter(suppressWarningsStrings::contains)
+                .findAny()
+                .get();
```

Comment:
I wonder about refactoring to avoid repeating this logic between the `disjoint` call above and here.

Reply:
Good idea, fixed in 7c5c0d5e6d83e17d8b418999a05863e67ec6d7d0